src/cryptonote_core/blockchain.cpp

```diff
@@ -1176,9 +1176,18 @@ bool Blockchain::validate_miner_transaction(const block& b, size_t cumulative_bl
     }
   }
 
-  std::vector<uint64_t> last_blocks_weights;
-  get_last_n_blocks_weights(last_blocks_weights, CRYPTONOTE_REWARD_BLOCKS_WINDOW);
-  if (!get_block_reward(epee::misc_utils::median(last_blocks_weights), cumulative_block_weight, already_generated_coins, base_reward, version))
+  uint64_t penalty_weight;
```

Comment:
Bikeshedding, but this variable naming sounds a little unintuitive. How about something like `effective_weight_median`?

Reply:
Still sounds awkward, but whatever.

Reply:
get_block_reward has it named median_weight, so it's a median there anyway. I renamed it the same.